src/markdown.ts

```diff
@@ -358,23 +359,26 @@ function getHead(data: FrontMatter, options: ParseOptions): string | null {
   return head;
 }
 
-function getHeader(data: FrontMatter, options: ParseOptions): string | null {
-  return getHtml("header", data, options);
+function getHeader(title: string | null, data: FrontMatter, options: ParseOptions): string | null {
+  return getHtml("header", title, data, options);
 }
 
-function getFooter(data: FrontMatter, options: ParseOptions): string | null {
-  return getHtml("footer", data, options);
+function getFooter(title: string | null, data: FrontMatter, options: ParseOptions): string | null {
+  return getHtml("footer", title, data, options);
 }
 
 function getHtml(
   key: "head" | "header" | "footer",
+  title: string | null,
   data: FrontMatter,
   {path, [key]: defaultValue}: ParseOptions
 ): string | null {
   return data[key] !== undefined
     ? data[key]
       ? String(data[key])
       : null
+    : typeof defaultValue === "function"
+    ? rewriteHtmlPaths(defaultValue(title, data, path), path)
     : defaultValue != null
     ? rewriteHtmlPaths(defaultValue, path)
```

Comment:
Would rather combine this like so:

```suggestion
    : defaultValue != null
    ? rewriteHtmlPaths(typeof defaultValue === "function" ? defaultValue(title, data, path) : defaultValue, path)
```